core/src/pageId.rs

```diff
@@ -0,0 +1,131 @@
+//! TODO
+
+use crate::{page::DEPTH, trie::KeyPath};
+use bitvec::prelude::*;
+
+pub type PageId = [u8; 32];
+
+pub const ROOT_PAGE_ID: [u8; 32] = [0; 32];
+
+/// Each Page is identified by a unique PageId,
+/// Root Page has id 0, its leftmost child 1 and so on.
+///
+/// There are 2^250 possible pages, so 250 bits are needed to represent all of them.
+///
+/// An array of bytes will be used as a container for the PageId,
+/// bytes will be treated as a 256-bit big-endian integer.
+///
+/// (TODO: treating them as little-endian would simply a lot offsets)
+///
+/// Given a KeyPath, it will be divided into chunks of 6 bits (sextet), forming
+/// an array of sextets. Each PageId will be constructed in the following manner:
+///
+/// page_ids[i] = (prev_page_id << d) + int(sextets[i]) + 1
+///
+/// The following struct implements an Iterator of PageIds over a KeyPath
+/// to lazily construct them as needed
+struct PageIdsIterator {
+    key_path: KeyPath,
```

Comment:
does the iterator need to own the path of just borrow it?

Reply:
Currently, it borrows it